### ISSUE TITLE
Read-only users are able to run mutations

### ISSUE THREAD

**Issue by robertrouse** [neo4j-graphql/neo4j-graphql#165]
I have a Neo4j Enterprise instance with a **read-only** user account. When passing the authentication header based on that account's credentials, I am **able to run mutations and write new data**. This should not be possible for accounts with the "reader" role.

**Comment by michaeldgraham** [comment]
Hey there, are you using authentication schema directives?

**Comment by robertrouse** [comment]
No. I didn't see anything about that in the docs. Based on your comment, I did find a blog post on it just now. I'll try adding @isAuthenticated to all my type defs. 

-Robert

**Comment by michaeldgraham** [comment]
Oh sorry, I deleted my comment because I mistakenly thought this post to be on the neo4j-graphql-js repo. Still, support for those auth directives isn't yet implemented in neo4j-graphql-js, though it should be soon. 

Sorry about the confusion.

**Comment by jexp** [comment]
Did you use the basic auth header
Can you share the output of list users which roles are listed for your user?

**Comment by robertrouse** [comment]
Yes, it uses {Authorization : "Basic (base64username:password)"} as the header. Here's my user list. I'm testing with the "biblereader" user. This is a fresh dev instance, so any part of my config is flexible for testing purposes.

![screen shot 2019-02-08 at 5 29 08 am](https://user-images.githubusercontent.com/1233286/52475865-87fffa00-2b62-11e9-8946-705d03a2338a.png)

**Comment by jexp** [comment]
Thanks a lot. I’ll investigate

**Comment by jexp** [comment]
Hi Robert,

sorry for the delay. I investigated it and you're right I could reproduce that the db provided by the server extension didn't adhere to the security rules.

**Comment by calinconstantinov** [comment]
Hi @jexp,

Is there any update on this issue?
